tests/test_basic_example_v2.py

```diff
@@ -0,0 +1,146 @@
+"""
+    Copyright 2021 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+# Note: These tests only function when the pytest output is not modified by plugins such as pytest-sugar!
+
+import importlib
+import logging
+import os
+import subprocess
+import sys
+import tempfile
+from types import ModuleType
+from typing import Optional, Sequence
+
+import pkg_resources
+import pytest
+from pkg_resources import DistributionNotFound
+
+# be careful not to import any core>=6 objects directly
+from inmanta import env
+from packaging import version
+
+CORE_VERSION: Optional[version.Version]
+"""
+Version of the inmanta-core package. None if it is not installed.
+"""
+
+try:
+    CORE_VERSION = version.Version(
+        pkg_resources.get_distribution("inmanta-core").version
+    )
+except DistributionNotFound:
+    CORE_VERSION = None
+
+
+pytestmark = pytest.mark.skipif(
+    CORE_VERSION is None or CORE_VERSION < version.Version("6.dev"),
+    reason="Skipping modules v2 tests for inmanta-core<6 (pre modules v2).",
+)
```

Comment:
This marker is not used anywhere?

Reply:
`pytest` itself interprets this. This is how to set a skip mark for an entire test module: https://docs.pytest.org/en/latest/how-to/skipping.html#skip-all-test-functions-of-a-class-or-module. But now that I read the documentation again I see that there might be a better way: `if condition: pytest.skip()` instead of `pytestmark = pytest.skipif(condition)`. I'll change this.

Reply:
Done